### PR TITLE
TST: stats: Loosen tolerance in some binomtest tests.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
See https://github.com/scipy/scipy/issues/14169 for background.

### CODE REVIEW

**Comment by mdhaber** [comment]
@WarrenWeckesser do you agree that this is probably happening when we look for the probability mass in the second tail - a small change in the evaluation of the binomial pmf is leading to a change in the integer at which the `cdf` (or `sf`) is calculated? (Does this make sense? If so, do you recall whether we checked this guess?)

**Comment by WarrenWeckesser** [comment]
@mdhaber, that is plausible, but I'll reserve judgement on whether or not that is *probably* what is happening.  For the test that triggered the failure, I took a look at the pmf values around the value that the binary search should have found as the starting point for the sum of the opposite tail,  and the values don't appear to be a borderline case.

**Comment by mdhaber** [comment]
So why do we think that 1e-11 is an acceptable error on this platform? Is it that the distribution `cdf`/`sf` is much less accurate on this platform?

**Comment by WarrenWeckesser** [comment]
> So why do we think that 1e-11 is an acceptable error on this platform? 

We face this question all over test suite; we don't have a rigorous definition of *acceptable error*.

Without easy access to the platforms where the relative errors are worse than that on the platform where the code was initially developed, it is tough to figure why the error is larger there.  In this case, we use boost for the pmf, cdf and sf calculations, but the behavior of boost might depend on compiler flags, the implementation of the underlying math library, and other platform-dependent factors.

**Comment by mdhaber** [comment]
Yes, I'm pretty familiar with these things. I have bumped many tolerances based on single-platform test failures before. But in this context, 1e-11 is above a threshold of what I'd immediately ignore, especially since a) we're comparing it to a p-value that is only ~1e-3 away from 1 and b) this is the first release of the new and improved `binomtest`. It's not _very_ concerning, but it's a large enough error that I'd like to be able to attribute it to something. I don't mean to block this tolerance bump; I guess I'll investigate on my own if I care enough.

**Comment by mdhaber** [comment]
Yes, this is due to inaccuracy of `binom` on ARM. I built master on an Ubuntu 18 64-bit Arm t4g.medium instance on EC2 and:
```
>>> from scipy.stats import binom
>>> k, n, p, y = (10079990, 21000000, 0.48, 10919991)
>>> binom.cdf(k, n, p) + binom.sf(n - y, n, p) - 0.9966892187965
-1.4906076373222277e-11
```
In that case, fine to loosen the test tolerances!

File: scipy/stats/tests/test_morestats.py [hunk 1]
```diff
@@ -778,13 +778,12 @@ class TestBinomTest:
...
     # binom.test(484, 967, p=0.48)
-    @pytest.mark.xfail_on_32bit("The large inputs make these tests "
-                                "sensitive to machine epsilon level")
+    #
     def test_two_sided_pvalues1(self):
         # These tests work on all OS's but fail on
         # Linux_Python_37_32bit_full due to numerical issues caused
         # by large inputs.
-        rtol = 1e-13  # aarch64 observed rtol: 3.5e-13
+        rtol = 1e-10  # aarch64 observed rtol: 3.5e-13
```
Comment (mdhaber) [inline scipy/stats/tests/test_morestats.py:786]:
```suggestion
        rtol = 1e-10  # arm64 observed rtol: 1.5e-11
```
?

File: scipy/stats/tests/test_morestats.py [hunk 2]
```diff
@@ -778,13 +778,12 @@ class TestBinomTest:
     # Expected results here are from R binom.test, e.g.
     # options(digits=16)
     # binom.test(484, 967, p=0.48)
-    @pytest.mark.xfail_on_32bit("The large inputs make these tests "
-                                "sensitive to machine epsilon level")
+    #
     def test_two_sided_pvalues1(self):
         # These tests work on all OS's but fail on
         # Linux_Python_37_32bit_full due to numerical issues caused
         # by large inputs.
```
Comment (mdhaber) [inline scipy/stats/tests/test_morestats.py:785]:
```suggestion
        # `tol` could be stricter on most architectures, but the value
        # here is limited by accuracy of `binom.cdf` for large inputs on
        # Linux_Python_37_32bit_full and aarch64
```

**Comment by tylerjereddy** [comment]
Thanks for working on this, I've added a backport label. It is the last "known" blocker for the `1.7.0` release candidate process (fingers crossed).

I won't get to building the wheels tonight anyway, but sounds like good chances of getting this merged tomorrow (Friday) based on the discussion.

**Comment by WarrenWeckesser** [comment]
Thanks @mdhaber, I committed your suggestions.

**Comment by mdhaber** [comment]
Thanks @WarrenWeckesser for fixing and @tylerjereddy for backporting.

**Pull request merged**